src/System.Net.WebSockets.Client/tests/ClientWebSocketOptionsTests.netcoreapp.cs

```diff
@@ -104,5 +104,28 @@ public void RemoteCertificateValidationCallback_Roundtrips()
                 }), new LoopbackServer.Options { UseSsl = true, WebSocketEndpoint = true });
             }
         }
+
+        [ConditionalTheory(nameof(WebSocketsSupported))]
+        [InlineData("ws://")]
+        [InlineData("wss://")]
+        public async Task NonSecureConnect_ConnectThruProxy_CONNECTisUsed(string connectionType)
+        {
+            if (PlatformDetection.IsWindows7)
+            {
+                return; // [ActiveIssue(27846)]
+            }
+
+            await LoopbackServer.CreateClientAndServerAsync(async proxyUri =>
+            {
+                using (var cws = new ClientWebSocket())
+                {
+                    cws.Options.Proxy = new WebProxy(proxyUri);
+                    try { await cws.ConnectAsync(new Uri(connectionType + Guid.NewGuid().ToString("N")), default); } catch { }
+                }
+            }, server => server.AcceptConnectionAsync(async connection =>
+            {
+                Assert.Contains("CONNECT", await connection.Reader.ReadLineAsync());
```

Comment:
Nit: this test will likely pass if there's something that goes wrong with the ConnectAsync such that the connection to the server is never made.  You might add a `bool connectionAccepted = false;` before the above `await LoopbackServer.CreateClientAndServerAsync`, then here set it to `true`, and then below just before the end of the test do `Assert.True(connectionAccepted)`.

Reply:
Will fix.